packages/eslint-plugin/src/rules/enforce-module-boundaries.ts

```diff
@@ -125,7 +126,7 @@ export default createESLintRule<Options, MessageIds>({
       noImportsOfE2e: 'Imports of e2e projects are forbidden',
       noImportOfNonBuildableLibraries:
         'Buildable libraries cannot import or export from non-buildable libraries',
-      noImportsOfLazyLoadedLibraries: `Imports of lazy-loaded libraries are forbidden`,
+      noImportsOfLazyLoadedLibraries: `Static imports of lazy-loaded libraries are forbidden.\n\nLibrary "{{targetProjectName}}" is lazy-loaded in "{{file}}"`,
```

Comment:
The library could be lazy loaded in several files. This should be a list like we have for circular dependencies.